sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/model/SqlStmt.kt

```diff
@@ -263,11 +264,21 @@ class SqlStmt private constructor(
       method.addParameter(parameter.build())
     }
 
+    var argsCodeBlock = CodeBlock.of("new String[0], ")
+
     if (arguments.isNotEmpty()) {
-      // Method body begins with local vars used during computation.
-      method.addStatement("\$T<String> args = new \$T<String>()", LIST_TYPE, ARRAYLIST_TYPE)
-          .addStatement("int currentIndex = 1")
-          .addStatement("\$1T query = new \$1T()", STRINGBUILDER_TYPE)
+      val hasNoStrings = arguments.map { it.argumentType.comparable?.dataType != TEXT }
```

Comment:
Ok I found the root cause for the test failures. I also need to check for `it.argumentType.comparable == null` since in case an `Object` is passed it it'll do some `instance of String` checks.

Example from `bind-object`:

```
public SqlDelightStatement some_select(Object arg1) {
  List<String> args = new ArrayList<String>();
  int currentIndex = 1;
  StringBuilder query = new StringBuilder();
  query.append("SELECT *, ");
  if (!(arg1 instanceof String)) {
    query.append(arg1);
  } else {
    query.append('?').append(currentIndex++);
    args.add((String) arg1);
  }
  query.append("\n"
          + "FROM test");
  return new SqlDelightStatement(query.toString(), args.toArray(new String[args.size()]), Collections.<String>singleton("test"));
}
```